digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala

```diff
@@ -1589,6 +1589,83 @@ object DataFixture {
     println("\n")
   }
 
+
+  private def isValidUnknownSpeedLimit(roadLink: RoadLink): Boolean = {
+    val roadLinkType = Seq(CycleOrPedestrianPath, PedestrianZone, TractorRoad, MotorwayServiceAccess, SpecialTransportWithoutGate, SpecialTransportWithGate, CableFerry)
+    val constructionType : Seq[ConstructionType] = Seq(UnderConstruction, Planned)
+
+    !((roadLinkType.contains(roadLink.linkType) || constructionType.contains(roadLink.constructionType)) && roadLink.administrativeClass == State)
+  }
+
+  def removeUnnecessaryUnknownSpeedLimits(): Unit = {
+    val dao = new OracleSpeedLimitDao(null, null)
+    val vvhClient = new VVHClient(dr2properties.getProperty("digiroad2.VVHRestApiEndPoint"))
+    val roadLinkService = new RoadLinkService(vvhClient, new DummyEventBus, new DummySerializer)
+
+    println("\nStart delete unknown speedLimits")
+    println(DateTime.now())
+
+    //Get All Municipalities
+    val municipalities: Seq[Int] = OracleDatabase.withDynSession { Queries.getMunicipalities  }
+
+    OracleDatabase.withDynTransaction {
+      municipalities.foreach { municipality =>
+        println(s"Obtaining all Road Links for Municipality: $municipality")
+
+        val unknownSpeedLimitByMunicipality = dao.getMunicipalitiesWithUnknown(municipality)
+        val roadLinks = roadLinkService.getRoadLinksAndComplementariesFromVVH(unknownSpeedLimitByMunicipality.toSet, false)
+
+        val filterRoadLinks = roadLinks.filterNot(isValidUnknownSpeedLimit)
+
+        if(filterRoadLinks.map(_.linkId).nonEmpty) {
```

Comment:
Can exist RoadLinks without LinkId's?
Can we switch the code to: 
```suggestion
        if(filterRoadLinks.nonEmpty) {
```